runhouse/resources/hardware/launcher_utils.py

```diff
@@ -34,12 +34,17 @@ def supported_providers():
 
     @classmethod
     def sky_secret(cls):
-        # TODO: is this guaranteed to be the default SkySecret name?
         secrets_name = "ssh-sky-key"
         try:
             sky_secret = rh.secret(secrets_name)
         except ValueError:
-            raise ValueError(f"No cluster secret found in Den with name {secrets_name}")
+            # Create a new default key pair required for the Den launcher and save it to Den
+            from runhouse import provider_secret
+
+            default_ssh_path, _ = generate_ssh_keys()
```

Comment:
@carolineechen wdyt of this approach? creating the default key pair only for the den launcher if we fail to find it in den (and not saving it down as the default key pair in local config, just ensuring it exists in Den)

Reply:
yea this works. we might even be able to skip writing down the secret to the file and just saving the values to Den/Vault if we don't need it locally but might need some testing, I think this is good for now